### PR TITLE
Add Ruby example

### PR DESCRIPTION
**Pull request by kaylareopelle**
During a recent OpenTelemetry spike, the Ruby agent team created an application that closely mirrors the example in New Relic's [Python OpenTelemetry Workshop](https://developer.newrelic.com/opentelemetry-masterclass/hands-on/workshop-python/). 

We'd like to share this example with the community.

### CODE REVIEW

**Review by tannalynn — APPROVED** [review]

**Comment by alanwest** [comment]
👋 Heya @kaylareopelle! Thanks for adding a Ruby example. You got this in draft, are you still working it?

**Comment by kaylareopelle** [comment]
Hi @alanwest! Thanks for pinging me! This is ready for review 😄

**Comment by reese-lee** [comment]
Hey @kaylareopelle! Thank you so much for putting this Ruby example together, and for separating it out into the directories per our Get Started Guide example in Java! I just have a few notes:

1. I know you mentioned OTel Ruby only has support for traces atm; I haven't gotten a chance to run the example yet, but does the tracing instrumentation produce what you would see [here](https://docs.newrelic.com/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/get-started/opentelemetry-tutorial-java/#traces) in the screen shots for traces?

2. I did see that it collects the same attributes, but can you rename them to `fibonacci.n` and `fibonacci.result`? 

3. The last thing (which I spaced on mentioning) is that we'd like all the examples to use the same port; could you update it to port 8080? 

Thanks again!

**Comment by kaylareopelle** [comment]
Hi @reese-lee! Circling back to this today! I've addressed your feedback here: 
* Set default port to 8080: ca5a33e5b6a75e8c9552a0b4e4b052bb540a4fe4 
* Use `fibonacci.n` and `fibonacci.result` as attribute names: 6a8a2987d76dea814797b0d541b452863fa75233 


Here's a Ruby-equivalent of the screenshot you shared: 
Screenshot 2023-06-07 at 4 30 20 PM

Please let me know if you need anything else!

File: README.md [hunk 1]
```diff
@@ -55,6 +55,8 @@ See [Contribute](#contribute) for how to request OpenTelemetry/New Relic interac
   - [OpenTelemetry SDK New Relic Config](./python): Two simple Python application demonstrating OpenTelemetry instrumentation, one with Flask auto-instrumentation and one without Flask. Configured for New Relic via the SDK.
 - Go
   - [OpenTelemetry SDK New Relic Config](./go): Simple Go applications demonstrating OpenTelemetry instrumentation.
+- Ruby
+  - [OpenTelemetry SDK New Relic Config](./ruby): A Sinatra application with auto- and manual-instrumentation.
```
Comment (kaylareopelle) [inline README.md:59]:
```suggestion
  - [OpenTelemetry SDK New Relic Config](./ruby): A Sinatra application with automatic and manual instrumentation.
```

File: getting-started-guides/ruby/instrumented/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+```
+* Make sure to use your [ingest license key](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/#license-key)
+* If your account is based in the EU, set the endpoint to: https://otlp.eu01.nr-data.net:4317
+
+3. Set this environment variable to name the demo app:
+```
+export OTEL_SERVICE_NAME=getting-started-ruby
+```
+
+4. Install your gems using Bundler
```
Comment (kaylareopelle) [inline getting-started-guides/ruby/instrumented/README.md:27]:
```suggestion
4. Install your gems using Bundler:
```

File: getting-started-guides/ruby/instrumented/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+```shell
+bundle install
+```
+
+5. Run the following command
```
Comment (kaylareopelle) [inline getting-started-guides/ruby/instrumented/README.md:33]:
```suggestion
5. Run the following command:
```

File: getting-started-guides/ruby/instrumented/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+```shell
+bundle exec rackup
+```
+
+4. To generate traffic, in a new terminal tab run the following command
```
Comment (kaylareopelle) [inline getting-started-guides/ruby/instrumented/README.md:39]:
```suggestion
4. To generate traffic, in a new terminal tab run the following command:
```

File: getting-started-guides/ruby/instrumented/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+* Make sure to use your [ingest license key](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/#license-key)
+* If your account is based in the EU, set the endpoint to: https://otlp.eu01.nr-data.net:4317
+
+3. Set this environment variable to name the demo app:
+```
+export OTEL_SERVICE_NAME=getting-started-ruby
+```
+
+4. Install your gems using Bundler
+
```
Comment (kaylareopelle) [inline getting-started-guides/ruby/instrumented/README.md:28]:
```suggestion
```

File: getting-started-guides/ruby/instrumented/README.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+```shell
+bundle install
+```
+
+5. Run the following command
+
```
Comment (kaylareopelle) [inline getting-started-guides/ruby/instrumented/README.md:34]:
```suggestion
```

File: getting-started-guides/ruby/instrumented/README.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
+# Getting Started Guide - Ruuby
```
Comment (reese-lee) [inline getting-started-guides/ruby/instrumented/README.md:1]:
```suggestion
# Getting Started Guide - Ruby
```

**Review by reese-lee — APPROVED** [review]
Hey Kayla! This looks pretty good to me. Unfortunately, I ran into issues with gems on my machine so I have not been able to run the app myself; if you can confirm that the trace data the app produces matches the trace data we're looking for [here](https://docs.newrelic.com/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/get-started/opentelemetry-tutorial-java/#view-data), I am good to get this merged!

**Comment by kaylareopelle** [comment]
@reese-lee - Sorry to hear you ran into trouble with gems! Here's a permalink to an app on staging with data from the instrumented app: https://staging.onenr.io/06vjAqpDnjP 

Things look quite similar. The only difference I see involves traces with errors: https://staging.onenr.io/0znQxBDeBRV 
The Ruby application doesn't have a nested span in this scenario. It has a nested span when the request completes successfully: https://staging.onenr.io/0PLREPoXewa

**Comment by reese-lee** [comment]
@kaylareopelle I think that's good, that might just be how it's handled in Ruby.

**Comment by kaylareopelle** [comment]
Great! In that case, please merge as you're able! (I don't have permission to do so)

**Comment by kaylareopelle** [comment]
Hi @reese-lee! Sorry about the delay here. I think that makes sense. Is this what you're looking for? https://staging.onenr.io/0kERzpxLlQr

**Review by alanwest — APPROVED** [review]

**Pull request merged**